pkgs/default.nix

```diff
@@ -216,8 +216,7 @@ pkgs: oldpkgs: {
         -Wall \
         -x c \
         "$textPath"
-    strip --strip-unneeded "$exe"
-  '';
+    '' + (if pkgs.stdenv.hostPlatform.isDarwin then "" else ''strip --strip-unneeded "$exe"'');
```

Comment:
You can use `optionalString` here if you like. And I think for readability it's might be better to embed this code instead of appending it:
```nix
  ${optionalString (!pkgs.stdenv.hostPlatform.isDarwin) /* sh */ ''
    strip --strip-unneeded "$exe"
  ''}
```